controlplane/kubeadm/controllers/kubeadm_control_plane_controller.go

```diff
@@ -295,23 +295,25 @@ func (r *KubeadmControlPlaneReconciler) updateStatus(ctx context.Context, kcp *c
 
 	replicas := int32(len(ownedMachines))
 	kcp.Status.Replicas = replicas
+	readyMachines := int32(0)
 
 	remoteClient, err := r.remoteClientGetter(ctx, r.Client, util.ObjectKey(cluster), r.scheme)
-	if err != nil && !apierrors.IsNotFound(errors.Cause(err)) {
+	if err != nil && !apierrors.IsNotFound(errors.Cause(err)) && !apierrors.IsTimeout(errors.Cause(err)) {
```

Comment:
If we have only `if err != nil` here, we can use a simple `else` condition for the `range`.

Inside the `if err != nil` block we can also do
```go
if cause := errors.Cause(err); !apierrors.IsNotFound(cause) && !apierrors.IsTimeout(cause) {
...
}
```

Reply:
done, thank you